system-int-test/test/cmr/system_int_test/search/variable/concept_retrieval_test.clj

```diff
@@ -0,0 +1,99 @@
+(ns cmr.system-int-test.search.variable.concept-retrieval-test
+  "Integration test for variable retrieval via the /concepts/:concept-id and
+  /concepts/:concept-id/:revision-id endpoints."
+  (:require
+   [cheshire.core :as json]
+   [clojure.test :refer :all]
+   [cmr.common.mime-types :as mt]
+   [cmr.mock-echo.client.echo-util :as e]
+   [cmr.system-int-test.system :as s]
+   [cmr.system-int-test.utils.index-util :as index]
+   [cmr.system-int-test.utils.ingest-util :as ingest]
+   [cmr.system-int-test.utils.metadata-db-util :as mdb]
+   [cmr.system-int-test.utils.search-util :as search]
+   [cmr.system-int-test.utils.variable-util :as variable]
+   [cmr.transmit.config :as transmit-config]))
+
+(use-fixtures
+ :each
+ (join-fixtures
+  [(ingest/reset-fixture {"provguid1" "PROV1"
+                          "provguid2" "PROV2"
+                          "provguid3" "PROV3"})
+   variable/grant-all-variable-fixture]))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;
+;; Retrieve by concept-id - general test
+;;
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(deftest retrieve-variable-by-concept-id
+  (let [var1-name "variable1"
+        var1-concept (variable/make-variable-concept {:Name var1-name})
+        var1 (ingest/ingest-concept var1-concept)
+        _ (index/wait-until-indexed)
+        var1-v2 (ingest/ingest-concept var1-concept)
```

Comment:
You need some way to say "yes this is the right metadata for revision 2". The way it is written right now there is no differentiation in the metadata between revision 1 and revision 2.

See lines 68 and 69 below to test that the latest revision for a concept is coming back:

`(= var1-name` ...

That test will pass even if revision 1 were incorrectly returned since both revision 1 and revision 2 have the same variable name.